code/modules/atmospherics/machinery/components/binary_devices/pressure_valve.dm

```diff
@@ -0,0 +1,176 @@
+/obj/machinery/atmospherics/components/binary/pressure_valve
+	icon_state = "pvalve_map-2"
+	name = "pressure valve"
+	desc = "An activable valve that let gas pass through if the pressure on the input side is higher than the set pressure."
+
+	can_unwrench = TRUE
+	shift_underlay_only = FALSE
+
+	///Amount of pressure needed before the valve for it to open
+	var/target_pressure = ONE_ATMOSPHERE
+	///Frequency for radio signaling
+	var/frequency = 0
+	///ID for radio signaling
+	var/id = null
+	///Connection to the radio processing
+	var/datum/radio_frequency/radio_connection
+	///Check if the gas is moving from one pipenet to the other
+	var/is_gas_flowing = FALSE
+
+	construction_type = /obj/item/pipe/directional
+	pipe_state = "pvalve"
+
+/obj/machinery/atmospherics/components/binary/pressure_valve/CtrlClick(mob/user)
+	if(can_interact(user))
+		on = !on
+		investigate_log("was turned [on ? "on" : "off"] by [key_name(user)]", INVESTIGATE_ATMOS)
+		update_icon()
+	return ..()
+
+/obj/machinery/atmospherics/components/binary/pressure_valve/AltClick(mob/user)
+	if(can_interact(user))
+		target_pressure = MAX_OUTPUT_PRESSURE
+		investigate_log("was set to [target_pressure] kPa by [key_name(user)]", INVESTIGATE_ATMOS)
+		update_icon()
+	return ..()
+
+/obj/machinery/atmospherics/components/binary/pressure_valve/Destroy()
+	SSradio.remove_object(src,frequency)
+	if(radio_connection)
+		radio_connection = null
+	return ..()
+
+/obj/machinery/atmospherics/components/binary/pressure_valve/update_icon_nopipes()
+	if(on && is_operational && is_gas_flowing)
+		icon_state = "pvalve_flow"
+	else if(on && is_operational && !is_gas_flowing)
+		icon_state = "pvalve_on"
+	else
+		icon_state = "pvalve_off"
+
+/obj/machinery/atmospherics/components/binary/pressure_valve/process_atmos()
+
+	if(!on || !is_operational)
+		return
+
+	var/datum/gas_mixture/air1 = airs[1]
+	var/datum/gas_mixture/air2 = airs[2]
+
+	if(air1.return_pressure() > target_pressure)
+		if(air1.pump_gas_to(air2, air1.return_pressure()))
```

Comment:
If we have pipe A and pipe B equal in volume, connected with this "pressure valve",
when pressure in A reach `target_pressure` it pump all gas from A to B ?

In my variant i think about just standard valve that opens self when pressure reached.
And closing self when pressure dropped.